### PR TITLE
CLOUDP-165343: [mongocli] Add support for initial sync via rest api

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-165343

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/usage/usage.go [hunk 1]
```diff
@@ -295,6 +295,7 @@ dbName and collection are required only for built-in roles.`
 	OrgOwnerID                                = "Unique 24-digit string that identifies the Atlas user to be granted the Org Owner role on the specified organization. Required if using API keys."
 	GovCloudRegionsOnly                       = "Flag that designates that the project uses only the AWS GovCloud region. Use this option only for Atlas for Government projects. If unspecified, the project uses only the AWS Standard region for AWS deployments. You can't deploy clusters across AWS GovCloud and AWS Standard regions in the same project."
 	ReclaimFreeSpaceTimestamp                 = "Timestamp in ISO 8601 format when the service reclaims the space. If not set, defaults to the current timestamp."
+	ResyncTimestamp                           = "Timestamp in ISO 8601 format when the starts to sync. If not set, defaults to the current timestamp."
```
Comment (sarahsimpers) [inline internal/usage/usage.go:298]:
```suggestion
	ResyncTimestamp                           = "Timestamp in ISO 8601 format when the sync starts. If not set, defaults to the current timestamp."
```

**Review by sarahsimpers — COMMENTED** [review]
Question and a couple of copy nits

File: internal/cli/opsmanager/clusters/resync.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	return telemetry.TrackAskOne(prompt, &opts.confirm)
+}
+
+// ResyncBuilder mongocli cloud-manager cluster(s) resync <clusterName> [--processName process1,process2...][--timestamp timestamp] [--force].
+func ResyncBuilder() *cobra.Command {
+	opts := &resyncOpts{}
+	cmd := &cobra.Command{
+		Use:   "resync <clusterName>",
+		Short: "Start an initial sync for a cluster or process.",
+		Long: `The MongoDB Agent checks whether the specified timestamp is later than the time of the last resync, and if confirmed, starts init sync on the secondary nodes in a rolling fashion.
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/opsmanager/clusters/resync.go:93]: I copy/paste this from https://www.mongodb.com/docs/ops-manager/v4.4/reference/cluster-configuration/#mongodb-instances (search for processes[n]. lastResync) so not sure if those docs are correct or not

File: internal/cli/opsmanager/clusters/resync.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+The MongoDB Agent waits until you ask the primary node to become the secondary with the rs.stepDown() method, and then starts init sync on this node.
+Warning: Use this method with caution. During initial sync, Automation removes the entire contents of the node’s dbPath directory. 
+See also: https://docs.mongodb.com/manual/tutorial/resync-replica-set-member/`,
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/opsmanager/clusters/resync.go]: done

File: internal/cli/opsmanager/clusters/resync.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+// ResyncBuilder mongocli cloud-manager cluster(s) resync <clusterName> [--processName process1,process2...][--timestamp timestamp] [--force].
+func ResyncBuilder() *cobra.Command {
+	opts := &resyncOpts{}
+	cmd := &cobra.Command{
+		Use:   "resync <clusterName>",
+		Short: "Start an initial sync for a cluster or process.",
+		Long: `The MongoDB Agent checks whether the specified timestamp is later than the time of the last resync, and if confirmed, the MongoDB Agent:
+1. Starts the initial sync on the secondary nodes in a rolling fashion
```
Comment (sarahsimpers) [inline internal/cli/opsmanager/clusters/resync.go:94]: Apologies, forgot a blank space is needed before this numbered list
```suggestion

1. Starts the initial sync on the secondary nodes in a rolling fashion
```

**Review by sarahsimpers — APPROVED** [review]
LGTM with one change (my fault, apologies!)

File: internal/cli/opsmanager/clusters/resync.go (hunk 4 above)
Comment (gssbzn) [reply internal/cli/opsmanager/clusters/resync.go:95]: done

**Pull request merged**